lib/livebook/utils.ex

```diff
@@ -368,4 +368,65 @@ defmodule Livebook.Utils do
     end)
     |> Enum.join("\n")
   end
+
+  # Fetches and formats system and node memory usage
+  def fetch_memory(node) do
+    node_memory = fetch_node_memory(node)
+    system_memory = fetch_system_memory()
+
+    %{node: node_memory, system: system_memory}
+  end
+
+  def fetch_memory() do
+    node_memory = %{total: %{value: 0}}
+    system_memory = fetch_system_memory()
+
+    %{node: node_memory, system: system_memory}
+  end
+
+  defp fetch_system_memory() do
+    :memsup.get_system_memory_data()
+    |> Enum.map(fn {k, v} ->
+      {k, %{unit: format_bytes(v), value: v}}
```

Comment:
I think we are good just keeping the value, we can always format when rendering, since it's a simple operation, wdyt?